src/Data/Aeson/TypeScript/Util.hs

```diff
@@ -148,9 +148,10 @@ mkInstance = InstanceD Nothing
 mkInstance = InstanceD
 #endif
 
-namesAndTypes :: Options -> [(Name, (Suffix, Var))] -> ConstructorInfo -> [(String, Type)]
-namesAndTypes options genericVariables ci = case constructorVariant ci of
-  RecordConstructor names -> zip (fmap ((fieldLabelModifier options) . lastNameComponent') names) (constructorFields ci)
+namesAndTypes :: ExtraTypeScriptOptions -> Options -> [(Name, (Suffix, Var))] -> ConstructorInfo -> [(String, Type)]
+namesAndTypes extraOptions options genericVariables ci = case constructorVariant ci of
+  RecordConstructor names -> let indiciesToDrop = foldr (\(i, recordFieldName) skipped -> if recordFieldName `elem` omitFields extraOptions then i:skipped else skipped) [] $ zip [(0 :: Int)..] $ fmap lastNameComponent' names
+                             in dropIndicies indiciesToDrop $ zip (fmap ((fieldLabelModifier options) . lastNameComponent') names) (constructorFields ci)
```

Comment:
This might be a little more idiomatic:

```suggestion
  RecordConstructor names -> fmap (\(recordFieldName, t) -> (fieldLabelModifier options recordFieldName, t)) 
                               $ filter (\(recordFieldName, _) -> recordFieldName `notElem` omitFields extraOptions) 
                               $ zip (fmap lastNameComponent' names) (constructorFields ci)                             
```

With fusion, this should also avoid traversing the list multiple times.